projects/go-lib/src/lib/components/go-table/go-table.component.html

```diff
@@ -1,26 +1,51 @@
 <div class="go-table-container" *ngIf="showTable">
-  <table class="go-table" *ngIf="hasData(); else noDataTable">
-    <thead>
-      <tr>
-        <th *ngFor="let col of columns" class="go-table__th" [ngClass]="columnClasses(col.field)" (click)="toggleSort(col.field)">
-          <ng-container *ngIf="!col.goTableHead; else columnHeaderOutlet">{{ col.title || col.field }}</ng-container>
-          <ng-template #columnHeaderOutlet>
-            <ng-container *ngTemplateOutlet="col.goTableHead"></ng-container>
-          </ng-template>
-        </th>
-      </tr>
-    </thead>
-    <tbody>
-      <tr *ngFor="let item of localTableConfig.tableData" class="go-table__tr">
-        <td *ngFor="let col of columns" class="go-table__td" width="{{ col.width || 'auto' }}">
-          <ng-container *ngIf="!col.goTableCell; else columnCellOutlet">{{ col.getFieldData(item) }}</ng-container>
-          <ng-template #columnCellOutlet>
-            <ng-container *ngTemplateOutlet="col.goTableCell; context: { $implicit: item }"></ng-container>
-          </ng-template>
-        </td>
-      </tr>
-    </tbody>
-  </table>
+  <div class="go-table__table">
+    <table class="go-table" *ngIf="hasData(); else noDataTable">
+      <thead>
+        <tr>
+          <th *ngFor="let col of columns" class="go-table__th" [ngClass]="columnClasses(col.field)" (click)="toggleSort(col.field)">
+            <ng-container *ngIf="!col.goTableHead; else columnHeaderOutlet">{{ col.title || col.field }}</ng-container>
+            <ng-template #columnHeaderOutlet>
+              <ng-container *ngTemplateOutlet="col.goTableHead"></ng-container>
+            </ng-template>
+          </th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr *ngFor="let item of setDisplayData()" class="go-table__tr">
+          <td *ngFor="let col of columns" class="go-table__td" width="{{ col.width || 'auto' }}">
+            <ng-container *ngIf="!col.goTableCell; else columnCellOutlet">{{ col.getFieldData(item) }}</ng-container>
+            <ng-template #columnCellOutlet>
+              <ng-container *ngTemplateOutlet="col.goTableCell; context: { $implicit: item }"></ng-container>
+            </ng-template>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+  <div class="go-table-toolbar go-table-toolbar--bottom" *ngIf="showPaging()">
+    <div class="go-table-page-size">
+      <select class="go-form__select" (change)="setPerPage($event)">
+        <ng-container *ngFor="let size of localTableConfig.paging.pageSizes">
+          <option value="{{ size }}">{{ size }}</option>
+        </ng-container>
+      </select>
+      <label class="go-form__label go-form__label--table-toolbar">Page Size</label>
```

Comment:
I think this would be more appropriate as an element selector inside of the `go-table-toolbar`. I think an easy way to look at it is:
```
block == component of shared UI
element == chunk inside the component that isn't shared
modifier == alterations or additions to things that get applied when the state changes.
```

If we think about it like that, this `go-table-toolbar` could realistically be a block/component _(not that it should be it's own component)_. Most of these other things are elements inside of that block. It might end up looking something like this:
```html
<div class="go-table-toolbar" *ngIf="showPaging()">
  <div class="go-form go-table-toolbar__page-size">
    <select class="go-form__select" (change)="setPerPage($event)">
      <option [ngValue]="size" *ngFor="let size of localTableConfig.paging.pageSizes">
        {{ size }}
      </option>
    </select>
    <label class="go-form__label go-table-toolbar__label">Page Size</label>
  </div>
  <div class="go-table-toolbar__totals">
    {{ outputResultsPerPage() }} /
    {{ localTableConfig.totalCount | number : '1.0-0' }} Results
  </div>
  <div class="go-table-page-controls">
    <go-icon class="go-table-page-controls__icon" [ngClass]="{ 'go-table-page-controls__icon--disabled': isFirstPage() }" (click)="setFirstPage()" icon="first_page"></go-icon>
    <go-icon class="go-table-page-controls__icon" [ngClass]="{ 'go-table-page-controls__icon--disabled': isFirstPage() }" (click)="prevPage()" icon="chevron_left"></go-icon>
    <go-icon class="go-table-page-controls__icon" [ngClass]="{ 'go-table-page-controls__icon--disabled': isLastPage() }" (click)="nextPage()" icon="chevron_right"></go-icon>
    <go-icon class="go-table-page-controls__icon" [ngClass]="{ 'go-table-page-controls__icon--disabled': isLastPage() }" (click)="setLastPage()" icon="last_page"></go-icon>
  </div>
</div>
```

Reply:
Yeah, that does make sense. I've change it to `go-table-toolbar__label`.